runtime/gc_base/ObjectAccessBarrier.cpp

```diff
@@ -1152,13 +1157,18 @@ MM_ObjectAccessBarrier::copyObjectFieldsToFlattenedArrayElement(J9VMThread *vmTh
 void
 MM_ObjectAccessBarrier::copyObjectFieldsFromFlattenedArrayElement(J9VMThread *vmThread, J9ArrayClass *arrayClazz, j9object_t destObject, J9IndexableObject *arrayRef, I_32 index)
 {
-	UDATA const objectHeaderSize = J9VMTHREAD_OBJECT_HEADER_SIZE(vmThread);
+	UDATA elementStartOffset = J9VMTHREAD_OBJECT_HEADER_SIZE(vmThread);
 	U_8 *elementAddress = (U_8*)indexableEffectiveAddress(vmThread, arrayRef, index, J9ARRAYCLASS_GET_STRIDE((J9Class *) arrayClazz));
 	IDATA elementOffset = (elementAddress - (U_8*)arrayRef);
 	J9Class *elementClazz = J9GC_J9OBJECT_CLAZZ_THREAD(destObject, vmThread);
 	Assert_MM_true(J9_IS_J9CLASS_VALUETYPE(elementClazz));
 	Assert_MM_true(elementClazz == arrayClazz->leafComponentType);
-	copyObjectFields(vmThread, elementClazz, (j9object_t) arrayRef, elementOffset, destObject, objectHeaderSize);
+
+	if (J9CLASS_HAS_4BYTE_PREPADDING(elementClazz)) {
+		elementStartOffset += sizeof(U_32);
+	}
```

Comment:
This pattern repeats a few times.

Consider introducing macros like J9CLASS_PREPADDING_SIZE, in which case this would convert into one liner:
elementStartOffset += J9CLASS_PREPADDING_SIZE(class)

Another macro to consider is J9VMTHREAD_OBJECT_HEADER_WITHPREPADDING_SIZE